lib/proxy/http_download.rb

```diff
@@ -3,35 +3,55 @@
 module Proxy
   class HttpDownload < Proxy::Util::CommandTask
     include Util
-    DEFAULT_READ_TIMEOUT = 60
     DEFAULT_CONNECT_TIMEOUT = 10
-    DEFAULT_DNS_TIMEOUT = 10
 
     def initialize(src, dst, read_timeout = nil, connect_timeout = nil, dns_timeout = nil, verify_server_cert = false)
       @dst = dst
-      wget = which("wget")
-      read_timeout ||= DEFAULT_READ_TIMEOUT
-      dns_timeout ||= DEFAULT_CONNECT_TIMEOUT
       connect_timeout ||= DEFAULT_DNS_TIMEOUT
+      args = [which('curl')]
 
-      args = ["--connect-timeout=#{connect_timeout}",
-              "--dns-timeout=#{dns_timeout}",
-              "--read-timeout=#{read_timeout}",
-              "--tries=3",
-              "--timestamping", # turn on timestamping to prevent redownloads
-              "--no-if-modified-since", # but use HTTP HEAD instead If-Modified-Since
-              "-nv", src.to_s, "-O", dst.to_s]
-      args.unshift("--no-check-certificate") unless verify_server_cert
-      args.unshift(wget)
-      super(args)
+      # no cert verification if set
+      args << "--insecure" unless verify_server_cert
+      # print nothing
+      args << "--silent"
+      # except errors
+      args << "--show-error"
+      # timeout (others were supported by wget but not by curl)
+      args << ["--connect-timeout", connect_timeout.to_s]
```

Comment:
I'm not a fan of adding arrays and then flattening it. How about:
```suggestion
      args += ["--connect-timeout", connect_timeout.to_s]
```

Reply:
Good point, makes sense.